### DISCUSSION TITLE
Can I insert a tab in the replace text ?  eg. -or '$1\t$2'

### DISCUSSION THREAD

**Discussion by Binger-cn** [BurntSushi/ripgrep#1953] [General] [answered]
Now using '$1\t$2' results in :    [group1]\t[group2]    , not     [group1]       [group2]

**Comment by BurntSushi** [comment, marked as answer]
You can, but you have to insert it literally. The replacement directive doesn't support escape sequences. Inserting it literally depends on your terminal/shell environment. In my environment, if I hit `CTRL-v`, release it, and then press `TAB`, it will insert a literal tab character.

**Comment by condekind** [comment]
An option that allows you to use `\t`: add `$` before your replace string, which **must** use single quotes, e.g., `--replace=$'$1\t$2'`
Full example:
```bash
$ cat example.txt
foo="this is foo"    hello _bar="this is bar" bye
    hmm hi foo="more foo" but   bar="more bar" ok...
```
```bash
$ rg -P '^.*foo="(.*)".*bar="(.*)".*$' --replace=$'\n${1}\t${2}\n\n'
./example.txt:
this is foo	this is bar

./example.txt:
more foo	more bar


```
As shown above, you can also use `\n` and other similar escapes (`\0` is also quite useful)
